kale/loaddata/video_multi_domain.py

```diff
@@ -91,29 +92,67 @@ def prepare_data_loaders(self):
+                self._rgb_target_by_split["train"] = get_class_subset(
+                    self._rgb_target_by_split["train"], self.class_ids
+                )
+                self._rgb_target_by_split["valid"] = get_class_subset(
+                    self._rgb_target_by_split["valid"], self.class_ids
+                )
 
             logging.debug("Load RGB Test")
-            self._rgb_source_by_split["test"] = self._source_access_dict["rgb"].get_test(self.class_ids)
-            self._rgb_target_by_split["test"] = self._target_access_dict["rgb"].get_test(self.class_ids)
+            self._rgb_source_by_split["test"] = self._source_access_dict["rgb"].get_test()
+            self._rgb_target_by_split["test"] = self._target_access_dict["rgb"].get_test()
+            if self.class_ids is not None:
+                self._rgb_source_by_split["test"] = get_class_subset(self._rgb_source_by_split["test"], self.class_ids)
+                self._rgb_target_by_split["test"] = get_class_subset(self._rgb_target_by_split["test"], self.class_ids)
 
         if self.flow:
             logging.debug("Load flow train and val")
             (self._flow_source_by_split["train"], self._flow_source_by_split["valid"]) = self._source_access_dict[
                 "flow"
-            ].get_train_val(self._val_split_ratio, self.class_ids)
+            ].get_train_val(self._val_split_ratio)
+            if self.class_ids is not None:
+                self._flow_source_by_split["train"] = get_class_subset(
+                    self._flow_source_by_split["train"], self.class_ids
+                )
+                self._flow_source_by_split["valid"] = get_class_subset(
+                    self._flow_source_by_split["valid"], self.class_ids
+                )
 
             (self._flow_target_by_split["train"], self._flow_target_by_split["valid"]) = self._target_access_dict[
                 "flow"
-            ].get_train_val(self._val_split_ratio, self.class_ids)
+            ].get_train_val(self._val_split_ratio)
+            if self.class_ids is not None:
+                self._flow_target_by_split["train"] = get_class_subset(
+                    self._flow_target_by_split["train"], self.class_ids
+                )
+                self._flow_target_by_split["valid"] = get_class_subset(
+                    self._flow_target_by_split["valid"], self.class_ids
+                )
 
             logging.debug("Load flow Test")
-            self._flow_source_by_split["test"] = self._source_access_dict["flow"].get_test(self.class_ids)
-            self._flow_target_by_split["test"] = self._target_access_dict["flow"].get_test(self.class_ids)
+            self._flow_source_by_split["test"] = self._source_access_dict["flow"].get_test()
+            self._flow_target_by_split["test"] = self._target_access_dict["flow"].get_test()
+            if self.class_ids is not None:
+                self._flow_source_by_split["test"] = get_class_subset(
```

Comment:
Could we put this function in the `DatasetAccess`? Is it feasible?
For example,
self._flow_source_by_split["test"] = self._flow_source_by_split["test"].get_class_subset(self.class_ids)

Reply:
Yes agreed, that will be better. Thanks, I'll update it.

Reply:
Ohh actually it may not be feasible. So, `self._flow_source_by_split["test"]` is actually a normal `torch.Dataset` and not a DatasetAccess object. 

If this function was to be added in the class, it could be a static function, and used something like: `DatasetAccess. get_class_subset(dataset, class_ids)`

Reply:
Ohh, so it may not be feasible as is, since `self._flow_source_by_split["test"]` is a simple` torch.Dataset` object and not a `DatasetAccess` object.

The function can be put inside DatasetAccess class as a static function, so it will be used like: `dataset = DatasetAccess.get_class_subset(dataset, class_ids)`. Would this be appropriate?

Reply:
Oh, I ignore the type. No need to change. That seems a little weird :) The current one is okay but it only has a few lines in `kale/utils/class_subset.py`. Do we have any other choice to put this function? We can keep the current now and discuss with Haiping at the meeting. Thanks!